src/main/java/teammates/common/datatransfer/InstructorAttributes.java

```diff
@@ -325,17 +315,20 @@ public boolean hasTutorPrivileges() {
     }
 
     /**
-     * pre-condition: instructorPrivilegesAsText and privileges should be non-null
+     * Compares this instructor with another instructor. JsonParser is used
+     * instead of {@code this.getJsonString().equals(instructor.getJsonString)}
+     * so that the comparison ignores the order of key-value pairs in the json
+     * strings.
```

Comment:
Don't mention anything about implementation details in header comment. Explain about `JsonParser` in the method body itself.